WordPress/src/main/java/org/wordpress/android/networking/GlideHeaderLoader.kt

```diff
@@ -0,0 +1,57 @@
+package org.wordpress.android.networking
+
+import android.util.Base64
+
+import com.bumptech.glide.load.Options
+import com.bumptech.glide.load.model.GlideUrl
+import com.bumptech.glide.load.model.Headers
+import com.bumptech.glide.load.model.LazyHeaders.Builder
+import com.bumptech.glide.load.model.ModelLoader
+import com.bumptech.glide.load.model.stream.BaseGlideUrlLoader
+
+import org.wordpress.android.fluxc.network.HTTPAuthManager
+import org.wordpress.android.fluxc.network.UserAgent
+import org.wordpress.android.fluxc.network.rest.wpcom.auth.AccessToken
+import org.wordpress.android.fluxc.utils.WPUrlUtils
+
+import java.io.InputStream
+
+import javax.inject.Inject
+
+/**
+ * Glide Header loader which adds an authorization header and a user agent header to requests.
+ */
+class GlideHeaderLoader @Inject constructor(
+    modelLoader: ModelLoader<GlideUrl, InputStream>,
+    private val mAccessToken: AccessToken,
+    private val mHttpAuthManager: HTTPAuthManager,
+    private val mUserAgent: UserAgent
+) : BaseGlideUrlLoader<GlideUrl>(modelLoader) {
+    override fun getHeaders(url: GlideUrl?, width: Int, height: Int, options: Options?): Headers? {
+        var headerBuilder = Builder()
+
+        url?.let {
```

Comment:
The existing `ImageLoader`-based implementation also has this step that makes sure WordPress.com URLs are `https` wrapped:

https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/1a2a3af600a05f8800d469dedb90240bb792efa6/fluxc/src/main/java/org/wordpress/android/fluxc/tools/FluxCImageLoader.java#L48-L50

Normally all WordPress.com URLs are `https`, but sometimes someone might manually edit a link or something and we'lll try to load through `http` and ignore the custom header work.

We should probably re-use that here too.